### PR TITLE
Bump Go modules

### PR DESCRIPTION
**Pull request by sjberman**
Updates all Go modules. Temporary replacement for smi-controller-sdk, due to issue: https://github.com/servicemeshinterface/smi-controller-sdk/issues/24.

The fork that's pointed to simply removes the printer package, which allows us to update our version of controller-runtime.

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/nginx-service-mesh/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [ ] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [x] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,9 +2,11 @@ module github.com/nginxinc/nginx-service-mesh
 
 go 1.19
 
+replace github.com/servicemeshinterface/smi-controller-sdk => github.com/sjberman/smi-controller-sdk v0.0.0-20221228160702-f642bac3bdf9
```
Comment (jbyers19) [inline go.mod:5]: Why is this necessary?

Comment (sjberman) [reply go.mod:5]: See the PR description.

Comment (jbyers19) [reply go.mod:5]: Sorry, haven't finished my coffee yet. 😄

**Review by jbyers19 — APPROVED** [review]

**Review by avahahn — APPROVED** [review]
